packages/astro/test/i18n-routing.test.js

```diff
@@ -1156,6 +1177,15 @@ describe('[SSR] i18n routing', () => {
 			let response = await app.render(request);
 			expect(response.status).to.equal(404);
 		});
+
+		it('should only match the default locale string in URLs where it is actually used as a locale', async () => {
+			// It should _not_ match `/en` from `/spanish/enigma` 
+			// and should respond with a 200
+			let request = new Request('http://example.com/new-site/spanish/enigma');
+			let response = await app.render(request);
+			expect(response.status).to.equal(200);
+			expect(await response.text()).includes('Astro Enigma');
```

Comment:
I believe you'd want to add an assertion for "Current Locale: es", not "Astro Enigma"